components/formats-gpl/src/loci/formats/in/LIFReader.java

```diff
@@ -1947,7 +2020,7 @@ else if (unit.equals("m")) {
             if (ms.sizeZ == 1) {
               ms.sizeZ = len;
               bytesPerAxis.put(nBytes, "Z");
-              physicalSizeZ = (physicalLen * len) / (len - 1);
+              physicalSizeZ = physicalLen;
```

Comment:
This was detected because for non-equal x/y dimension sizes the bioformats importer assumed the pixels were not square anymore. If you read in e.g. a 1024*512 pixel image and the pixels are not square according to ome-xml metadata, this should be a failed test case. I can provide lif files if needed.